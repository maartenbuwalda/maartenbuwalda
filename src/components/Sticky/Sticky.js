import React from 'react'
import PropTypes from 'prop-types'
import styled from 'styled-components'
// import colors from '../../constants/colors'

const StickyWrapper = styled.div`
  position: sticky;
  top: 0;
  z-index: 99;
`

const Sticky = ({ children }) => {
  return (
    <StickyWrapper>
      {children}
    </StickyWrapper>
  )
}

Sticky.propTypes = {
  children: PropTypes.node,
}

export default Sticky
