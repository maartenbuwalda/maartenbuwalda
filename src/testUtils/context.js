const context = {
  theme: {},
  updateTheme: jest.fn(),
}

export default context
