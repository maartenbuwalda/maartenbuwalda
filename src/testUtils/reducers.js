export const reducerReturnsInitialState = (reducer, initialState) => {
  it('should succeed: return the initialState', () => {
    expect(reducer(undefined, {})).toEqual(initialState)
  })
}
