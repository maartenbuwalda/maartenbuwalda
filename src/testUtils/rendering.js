import toJson from 'enzyme-to-json'
import { shallow, mount } from 'enzyme'
import configureStore from 'redux-mock-store'
import 'jest-styled-components'

export const componentRenders = component => {
  it('renders correctly', () => {
    const shallowed = shallow(component)
    expect(toJson(shallowed)).toMatchSnapshot()
  });
}

export const connectedComponentRenders = component => {
  const middlewares = []
  const mockStore = configureStore(middlewares)
  const initialState = {}
  const store = mockStore(initialState)
  
  it('renders correctly', () => {
    const shallowed = shallow(component, store)
    expect(toJson(shallowed)).toMatchSnapshot()
  })
}
