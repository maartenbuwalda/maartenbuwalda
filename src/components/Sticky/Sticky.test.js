import React from 'react'
import Sticky from './Sticky'
import { componentRenders } from '../../testUtils/rendering'

describe('Sticky component', () => {
  const component = <Sticky>test</Sticky>
  componentRenders(component)
})
