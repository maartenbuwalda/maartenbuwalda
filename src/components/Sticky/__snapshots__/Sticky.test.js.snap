// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sticky component renders correctly 1`] = `
<styled.div>
  test
</styled.div>
`;
